src/patternfly/layouts/Page/styles.scss

```diff
@@ -1,110 +1,307 @@
+      // Ensure button does not adjust positioning
+      margin-right: var(--pf-l-page__header-brand-link--MarginLeft);
+      margin-left: calc(var(--pf-l-page__header-sidebar-toggle--Padding) * -1);
+      font-size: var(--pf-l-page__header-sidebar-toggle--FontSize);
+      line-height: 1;
+      border: none;
+    }
+
+    // Brand
     &-brand {
-      flex: 0 0 var(--pf-l-page__sidebar--md--Width);
+      align-items: flex-end;
+      padding-top: var(--pf-l-page__header-brand--PaddingTop);
+      padding-bottom: var(--pf-l-page__header-brand--PaddingBottom);
+
+      @media (min-width: $pf-global--breakpoint--md) {
+        flex: 0 0 var(--pf-l-page__header-brand--Width--lg);
+        align-items: flex-end;
+        align-self: stretch;
+        padding-right: var(--pf-l-page__header-brand--PaddingRight);
+        padding-left: var(--pf-l-page__header-brand--PaddingLeft);
+        // Offset brand by padding values
+        margin-left: calc(var(--pf-l-page__header--PaddingLeft) * -1);
+        background-color: var(--pf-l-page__header-brand--BackgroundColor--md);
+      }
+
+      @media screen and (min-width: $pf-global--breakpoint--lg) {
+
+      }
+
+      // Link
+      &-link {
+        display: flex;
+        flex: 1;
+        align-items: center;
+        align-self: flex-end;
+        min-height: 40px;
+      }
+
+      .pf-m-patternfly-logo {
+        height: var(--pf-l-page__header-brand-logo--Height);
+      }
+    }
+
+    // Header navigation
+    &-nav {
+      flex: 1 0 100%;
+      order: 1;
+      // Set width value for container to scale properly
+      width: 100vw;
+      padding-left: var(--pf-l-page__header-nav--PaddingLeft);
+      // Header will may contain elements that are hidden at different breakpoints. 
+      // Using negative margins to offset header padding to allow those elements to layout naturally.
+      margin-right: calc(var(--pf-l-page__header--PaddingRight) * -1);
+      margin-left: calc(var(--pf-l-page__header--PaddingLeft) * -1);
+      overflow: hidden;
+      background-color: var(--pf-l-page__header-nav--BackgroundColor--sm);
+
+      @media screen and (min-width: $pf-global--breakpoint--lg) {
+        flex: 1;
+        order: initial;
+        width: auto;
+        padding: 0;
+        margin-right: var(--pf-l-page__header-nav--MarginRight--md);
+        margin-left: var(--pf-l-page__header-nav--MarginLeft--md);
+        background-color: transparent;
+      }
+    }
+
+    &-selector {
+      flex-shrink: 1;
+      align-items: center;
+      padding-top: var(--pf-l-page__header-selector--PaddingTop);
+      padding-bottom: var(--pf-l-page__header-selector--PaddingBottom);
+
+      @media screen and (min-width: $pf-global--breakpoint--md) {
+        padding-left: var(--pf-l-page__header-selector--PaddingLeft--md);
+      }
     }
 
     &-tools {
+      align-items: center;
       margin-left: auto; // to push it to the right
+
+      .pf-l-toolbar {
```

Comment:
This also deserves a comment, why are we positioning a single layout here. Also, why do we need a margin top since the container is`flex-end`

Reply:
Avatar has to position itself vertically without affecting the rest of layout. To do that and remain flexible, tools need equal margin-top and bottom values and the avatar needs to be outside `.pf-l-toolbar`.